### PR TITLE
Add Gentleman 65 SE Solderd PCB support

### PR DESCRIPTION
**Pull request by JJ48**
## Description

added support for Gentleman 65 SE Solderd PCB, made small bug fixes to gent 65 firmware, and added gentleman65 and gentleman65_se_s to the jkeys_design folder.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jkeys_design/gentleman65_se_s/config.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#define MATRIX_ROW_PINS { F0, F1, F4, F5, F6 }
+#define MATRIX_COL_PINS { D5, D3, D2, D1, D0, B7, B2, B3, D4, D6, D7, C7, C6, B6, B5, B4 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN F7
+#define RGBLED_NUM 14
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/jkeys_design/gentleman65_se_s/config.h:44]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/jkeys_design/gentleman65_se_s/gentleman65_se_s.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gentleman65_se_s.h"
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/jkeys_design/gentleman65_se_s/gentleman65_se_s.c:20]:
```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
```

File: keyboards/jkeys_design/gentleman65_se_s/gentleman65_se_s.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code_delay(KC_AUDIO_VOL_UP, 10);
+        } else {
+            tap_code_delay(KC_AUDIO_VOL_DOWN, 10);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/jkeys_design/gentleman65_se_s/gentleman65_se_s.c:30]:
```suggestion
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
Retargeted to `develop` as it involves a keyboard relocation.
Please update `data/mappings/keyboard_aliases.json` with the new location.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
This has some merge conflicts that need resolution before it can be merged.

**Comment by keyboard-magpie** [comment]
@JJ48 don't know if you saw Drashna's above.

**Review by zvecr — APPROVED** [review]

**Pull request merged**